spec/rubocop/cop/style/unneeded_condition_spec.rb

```diff
@@ -63,6 +63,15 @@
           RUBY
         end
       end
+
+      context 'when using modifier if' do
+        it 'registers an offense' do
+          expect_offense(<<-RUBY.strip_indent)
+            bar if bar
```

Comment:
I guess for those cases we should have a different message and a different correction.

Reply:
Or some dedicate lint cop. :-)

Reply:
> I guess for those cases we should have a different message and a different correction.

The message `Use double pipes `||` instead.` is generic enough to work here. Do have a recommendation for a different message?

How should the correction be different? If the the if statement is false, this will return `nil`. Without context the condition doesn't make sense in the first place. 


> Or some dedicate lint cop. :-)
This feels more like style than lint to me. Either way works though.

Reply:
Isn't it better to just drop the `if` here - `bar if bar` -> `bar`. That's what I was referring to. That's why I felt the message is confusing.

Reply:
Yes, you are correct. It took me a minute to realize that `nil` would be redundant after the correction. Given that the cop is named `UnneededCondition`, a message saying `This is an unneeded condition` would make sense to me.

Reply:
I added a simple message of `This condition is not needed.`. What do you think about combining the messages for other offenses? ```This condition is not needed. Use double pipes `||` instead.```

Reply:
I'd rather have them separate. There's also the point that often `a if a` is likely some mistake - perhaps you meant to type something else in such a scenario, right?